Sources/MongoSwift/Operations/ListDatabasesOperation.swift

```diff
@@ -17,28 +17,43 @@ public struct DatabaseSpecification: Codable {
 }
 
 /// Internal intermediate result of a ListDatabases command.
-internal enum ListDatabasesResults {
+public enum ListDatabasesResults {
     /// Includes the names and sizes.
     case specs([DatabaseSpecification])
 
     /// Only includes the names.
     case names([String])
 }
 
+/// Options for "listDatabases" operations
+public struct ListDatabasesOptions: Encodable {
+    /// Specifies whether to only return databases for which the user has privileges,
+    /// if false MongoDB will return an error for unauthorized users.
+    public let authorizedDatabases: Bool
```

Comment:
let's make this a `var`. that allows users to build options up programmatically like:
```swift
var opts = ListDatabasesOptions()
opts.authorizedDatabases = true
```